### PR TITLE
Moving PrincipalServerEvent from lib-admin-ui to app-users #298

### PR DESCRIPTION
**Pull request by ashklianko**
-Adjusted server event classes to be compatible with parent ones in lib-admin-ui
-Added UsersServerEventsTranslator to convert EventJson to PrincipalServerEvent

### CODE REVIEW

**Comment by ashklianko** [comment]
Requires https://github.com/enonic/lib-admin-ui/pull/1215

File: src/main/resources/assets/js/app/event/PrincipalServerChange.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export class PrincipalServerChangeBuilder
+    extends NodeServerChangeBuilder {
+
+    build(): PrincipalServerChange {
+        return new PrincipalServerChange(this);
+    }
+
+    getPathPrefix(): string {
+        return '/identity';
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/event/PrincipalServerChange.ts]: Make this a static property in `PrincipalServerChange` and use it here

File: src/main/resources/assets/js/app/event/PrincipalServerChangeItem.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export class PrincipalServerChangeItem
+    extends NodeServerChangeItem {
+
+    constructor(builder: PrincipalServerChangeItemBuilder) {
+        super(builder);
+    }
+
+    protected processPath(path: string): string {
+        return path.substr('/identity'.length);
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/event/PrincipalServerChangeItem.ts]: Use static property from `PrincipalServerChange`

File: src/main/resources/assets/js/app/event/PrincipalServerEvent.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export class PrincipalServerEvent
+    extends NodeServerEvent {
+
+    constructor(change: PrincipalServerChange) {
+        super(change);
+    }
+
+    static is(eventJson: NodeEventJson): boolean {
+        return eventJson.data.nodes.some(node => node.path.indexOf('/identity') === 0);
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/event/PrincipalServerEvent.ts]: Use static property from `PrincipalServerChange`

File: src/main/resources/assets/js/app/event/ReportServerChange.ts [hunk 4]
```diff
@@ -1,55 +1,32 @@
...
-    constructor(type: NodeServerChangeType, changeItems: ReportServerChangeItem[], newReportPaths?: string[]) {
-        super(type, changeItems, newReportPaths);
+    build(): ReportServerChange {
+        return new ReportServerChange(this);
     }
 
-    getChangeType(): NodeServerChangeType {
-        return this.type;
+    getPathPrefix(): string {
+        return '/reports';
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/event/ReportServerChange.ts]: Same here - make this a static property in `ReportServerChange`

File: src/main/resources/assets/js/app/event/ReportServerChangeItem.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+export class ReportServerChangeItem
+    extends NodeServerChangeItem {
+
+    constructor(builder: ReportServerChangeItemBuilder) {
+        super(builder);
+    }
+
+    protected processPath(path: string): string {
+        return path.substr('/reports'.length);
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/event/ReportServerChangeItem.ts]: Use static property from `ReportServerChange`

**Pull request merged**